enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/jmx/MBeanRegistrator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.choreo.connect.enforcer.jmx;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.choreo.connect.enforcer.commons.logging.ErrorDetails;
+import org.wso2.choreo.connect.enforcer.commons.logging.LoggingConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+
+/**
+ * The class which is responsible for registering MBeans.
+ */
+public class MBeanRegistrator {
+    private static final Logger logger = LogManager.getLogger(MBeanRegistrator.class);
+    private static List<ObjectName> mBeans = new ArrayList<>();
+
+    private static final String SERVER_PACKAGE = "org.wso2.choreo.connect.enforcer";
+
+    private MBeanRegistrator() {
+    }
+
+    /**
+     * Registers an object as an MBean with the MBean server.
+     *
+     * @param mBeanInstance - The MBean to be registered as an MBean.
+     */
+    public static void registerMBean(Object mBeanInstance) throws RuntimeException {
+
+        if (JMXUtils.isJMXMetricsEnabled()) {
+            String className = mBeanInstance.getClass().getName();
+            if (className.indexOf('.') != -1) {
+                className = className.substring(className.lastIndexOf('.') + 1);
```

Comment:
Do we need "." character to be considered here?

Reply:
https://github.com/wso2/carbon-kernel/blob/master/core/src/main/java/org/wso2/carbon/kernel/jmx/MBeanRegistrator.java#L52
We are only consider the class name part from the full name of the class and prefix with  `"org.wso2.choreo.connect.enforcer"`
Ex: 
`org.wso2.choreo.connect.enforcer.metrics.jmx.impl.ExtAuthMetrics` -> `org.wso2.choreo.connect.enforcer.ExtAuthMetrics`